src/game.cpp

```diff
@@ -414,22 +417,57 @@ PointedThing getPointedThing(Client *client, v3f player_position,
 						result.node_abovesurface = np_above;
 						mindistance = distance;
 
-						hilightboxes.clear();
-
-						if (!g_settings->getBool("enable_node_highlighting")) {
-							for (std::vector<aabb3f>::const_iterator
-									i2 = boxes.begin();
-									i2 != boxes.end(); ++i2) {
-								aabb3f box = *i2;
-								box.MinEdge += npf + v3f(-d, -d, -d) - intToFloat(camera_offset, BS);
-								box.MaxEdge += npf + v3f(d, d, d) - intToFloat(camera_offset, BS);
-								hilightboxes.push_back(box);
-							}
+						selectionboxes->clear();
+						for (std::vector<aabb3f>::const_iterator
+								i2 = boxes.begin();
+								i2 != boxes.end(); ++i2) {
+							aabb3f box = *i2;
+							box.MinEdge += v3f(-d, -d, -d);
+							box.MaxEdge += v3f(d, d, d);
+							selectionboxes->push_back(box);
 						}
+						hud->setHighlightedPos(npf, camera_offset);
+						hud->setSelectionMeshScale(v3f(1.0, 1.0, 1.0));
```

Comment:
Can you put a note somewhere explaining under what circumstances the selection mesh scale will vary?